Tools/CISupport/ews-build/factories.py

```diff
@@ -97,6 +97,27 @@ def __init__(self, platform, configuration=None, architectures=None, triggers=No
         self.addStep(ApplyWatchList())
 
 
+class SmartPointerStaticAnalyzerFactory(factory.BuildFactory):
+    findModifiedLayoutTests = False
+    branches = None
+
+    def __init__(self, platform, configuration=None, architectures=None, buildOnly=True, triggers=None, triggered_by=None, remotes=None, additionalArguments=None, checkRelevance=False, **kwargs):
+        factory.BuildFactory.__init__(self)
+        self.addStep(ConfigureBuild(platform=platform, configuration=configuration, architectures=architectures, buildOnly=buildOnly, triggers=triggers, triggered_by=triggered_by, remotes=remotes, additionalArguments=additionalArguments))
+        if checkRelevance:
+            self.addStep(CheckChangeRelevance())
+        self.addStep(ValidateChange(branches=self.branches))
```

Comment:
Do we need to pass branches (and initialize branches = None) at all, as self.branches is None and the default value of branches is also None.

Reply:
Good point. This was an artifact of following the `Factory` class initialization